pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -186,22 +179,22 @@ func (m *serviceAffinityMetadata) clone() *serviceAffinityMetadata {
 }
 
 type podAffinityMetadata struct {
-	topologyPairsAntiAffinityPodsMap *topologyPairsMaps
+	topologyPairsAntiAffinityPodsMap topologyPairToPods
 	// A map of topology pairs to a list of Pods that can potentially match
 	// the affinity terms of the "pod" and its inverse.
-	topologyPairsPotentialAffinityPods *topologyPairsMaps
+	topologyPairsPotentialAffinityPods topologyPairToPods
 	// A map of topology pairs to a list of Pods that can potentially match
 	// the anti-affinity terms of the "pod" and its inverse.
-	topologyPairsPotentialAntiAffinityPods *topologyPairsMaps
+	topologyPairsPotentialAntiAffinityPods topologyPairToPods
 }
 
 func (m *podAffinityMetadata) addPod(addedPod *v1.Pod, pod *v1.Pod, node *v1.Node) error {
 	// Add matching anti-affinity terms of the addedPod to the map.
-	topologyPairsMaps, err := getMatchingAntiAffinityTopologyPairsOfPod(pod, addedPod, node)
+	topologyPairToPods, err := getMatchingAntiAffinityTopologyPairsOfPod(pod, addedPod, node)
```

Comment:
We know our concern is map access,  so obtaining the map and then merging, instead of directly adding values to the existing map might give us slightly better performance.